libsolidity/analysis/TypeChecker.cpp

```diff
@@ -921,9 +921,17 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 				{
 					int numBits = type->numBits();
 					bool isSigned = type->isSigned();
+					string minValue;
+					string maxValue;
 					if (isSigned)
+					{
 						numBits--;
-					extension = ", which can hold values up to " + string((u256(1) << numBits) - 1);
+						minValue = "-" + string(u256(1) << numBits);
+					}
+					else
+						minValue = "0";
+					maxValue = string((u256(1) << numBits) - 1);
```

Comment:
Btw, in some places we use `boost::lexical_cast<string>()` and in others `string()`. Is there a reason for that?